From 2b3795805c8d1bd8873b046508777fa6e9a5c83d Mon Sep 17 00:00:00 2001
From: Jan Kara <jack@suse.cz>
Date: Tue, 9 Aug 2016 19:17:56 +0200
Subject: [PATCH] quotacheck: Use direct scanning also for ext4
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We mistakenly didn't use direct scanning through libext2fs for ext4
filesystem. Add ext4 to the list of filesystem libext2fs can handle.

Signed-off-by: Jan Kara <jack@suse.cz>
Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 quotacheck.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/quotacheck.c b/quotacheck.c
index 6f34cff..a6fe432 100644
--- a/quotacheck.c
+++ b/quotacheck.c
@@ -959,7 +959,10 @@ Please stop all programs writing to filesystem or use -m flag to force checking.
 start_scan:
 	debug(FL_VERBOSE | FL_DEBUG, _("Scanning %s [%s] "), mnt->me_devname, mnt->me_dir);
 #if defined(EXT2_DIRECT)
-	if (!strcmp(mnt->me_type, MNTTYPE_EXT2) || !strcmp(mnt->me_type, MNTTYPE_EXT3) || !strcmp(mnt->me_type, MNTTYPE_NEXT3)) {
+	if (!strcmp(mnt->me_type, MNTTYPE_EXT2) ||
+	    !strcmp(mnt->me_type, MNTTYPE_EXT3) ||
+	    !strcmp(mnt->me_type, MNTTYPE_NEXT3) ||
+	    !strcmp(mnt->me_type, MNTTYPE_EXT4)) {
 		if ((failed = ext2_direct_scan(mnt->me_devname)) < 0)
 			goto out;
 	}
-- 
2.7.4

